File: 13-Decorator/src/go/base_coffees.go
package main

type Espresso struct{}

func (e Espresso) GetDescription() string { return "Espresso" }
func (e Espresso) GetCost() float64       { return 2.00 }

type Latte struct{}

func (l Latte) GetDescription() string { return "Latte" }
func (l Latte) GetCost() float64       { return 3.00 }

File: 13-Decorator/src/rust/base_coffees.rs
use crate::coffee::Coffee;

pub struct Espresso;

impl Coffee for Espresso {
    fn get_description(&self) -> String { "Espresso".to_string() }
    fn get_cost(&self) -> f64 { 2.00 }
}

pub struct Latte;

impl Coffee for Latte {
    fn get_description(&self) -> String { "Latte".to_string() }
    fn get_cost(&self) -> f64 { 3.00 }
}
